src/ir_York.cpp

```diff
@@ -0,0 +1,340 @@
+/// @param[in] offset The starting index to use when attempting to decode the
+///   raw data. Typically/Defaults to kStartOffset.
+/// @param[in] nbits The number of data bits to expect.
+/// @param[in] strict Flag indicating if we should perform strict matching.
+/// @return A boolean. True if it can decode it, false if it can't.
+bool IRrecv::decodeYork(decode_results *results, uint16_t offset,
+                        const uint16_t nbits, const bool strict) {
+  if (results->rawlen < 2 * nbits + kYorkOverhead - offset)
+    return false;  // Too short a message to match.
+  if (strict && nbits != kYorkBits)
+    return false;
+
+  uint16_t used = 0;
+
+  used = matchGeneric(results->rawbuf + offset, results->state,
+                      results->rawlen - offset, nbits,
+                      kYorkHdrMark, kYorkHdrSpace,
+                      kYorkBitMark, kYorkOneSpace,
+                      kYorkBitMark, kYorkZeroSpace,
+                      kYorkBitMark, kDefaultMessageGap,
+                      false, _tolerance, kMarkExcess,
+                      false);  // LSB
+  if (used == 0) return false;  // We failed to find any data.
+
+  // Succes
+  results->decode_type = decode_type_t::YORK;
+  results->bits = nbits;
+
+  return true;
+}
+#endif  // DECODE_YORK
+
+//
+//
+/// Class constructor
+/// @param[in] pin GPIO to be used when sending.
+/// @param[in] inverted Is the output signal to be inverted?
+/// @param[in] use_modulation Is frequency modulation to be used?
+IRYorkAc::IRYorkAc(const uint16_t pin, const bool inverted,
+                         const bool use_modulation)
+      : _irsend(pin, inverted, use_modulation) { }
+
+/// Set up hardware to be able to send a message.
+void IRYorkAc::begin(void) { _irsend.begin(); }
+
+/// Get the raw state of the object, suitable to be sent with the appropriate
+/// IRsend object method.
+/// @return A copy of the internal state.
+uint8_t *IRYorkAc::getRaw(void) {
+  return _.raw;
+}
+
+/// Set the internal state from a valid code for this protocol.
+/// @param[in] new_code A valid code for this protocol.
+/// @param[in] length Length of the code in bytes.
+void IRYorkAc::setRaw(const uint8_t new_code[], const uint16_t length) {
+  std::memcpy(_.raw, new_code, length);
+}
+
+#if SEND_YORK
+/// Send the current internal state as an IR message.
+/// @param[in] repeat Nr. of times the message will be repeated.
+void IRYorkAc::send(const uint16_t repeat) {
+  _irsend.sendYork(getRaw(), kYorkBits, repeat);
+}
+#endif  // SEND_YORK
+
+/// Get the current operation mode setting.
+/// @return The current operation mode.
+uint8_t IRYorkAc::getMode(void) const {
+  return _.Mode;
+}
+
+/// Set the desired operation mode.
+/// @param[in] mode The desired operation mode.
+void IRYorkAc::setMode(const uint8_t mode) {
+  switch (mode) {
+    case kYorkFan:
+    case kYorkCool:
+    case kYorkHeat:
+    case kYorkDry:
+    case kYorkAuto:
```

Comment:
Technically, you don't need the `case kYorkAuto:`, as it will be covered by the the `default:` clause.